### PR TITLE
[DNM] doc/dev: explain the primary balancer scope and design

### PR DESCRIPTION
**Pull request by ljflores**
This commit was originally included in #44002, however, since this new page concerns a feature that hasn't been implemented yet, it should be merged after the Quincy feature freeze.

This commit was authored by @JoshSalomon.

Signed-off-by: Laura Flores <lflores@redhat.com>

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [x] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by ljflores** [comment]
unstale

**Comment by ljflores** [comment]
[balancer: balance reads](https://trello.com/c/is7l6BIb/812-balancer-balance-reads)

**Review by jdurgin — APPROVED** [review]
@zdover23 this one is ready, no need to backport since it's a new feature in reef

File: doc/dev/prim-balancer-design.rst [hunk 1]
```diff
@@ -0,0 +1,52 @@
+
+This document describes the requirements and HLD of the primary balancer for Ceph.
```
[the remark this answers is not in the document]
Comment (zdover23) [reply doc/dev/prim-balancer-design.rst:2]: https://github.com/ceph/ceph/pull/48920 expands "HLD" to "high-level design".

**Pull request merged**